tests/test_web.py

```diff
@@ -82,6 +83,24 @@ def email_body():
     return message.as_string()
 
 
+@pytest.fixture
+def rendered_body_template(environment, email_invoice_request_body):
+    env = Environment(
```

Comment:
I ask again: please don't put template building logic in the test code. Use a constant `str` for the expected output.